students/arun_nalla/session04/trigrams.py

```diff
@@ -0,0 +1,50 @@
+﻿#! usr/bin/env python 3
+#words = 'I wish I may I wish I might'.split()
+def trigram_dict (words):
+    trilist = {}
+    for i in range(len(words)-2):
+        a = tuple(words[i:i+2])
+        b = words[i+2]
+        if a not in trilist:
+            trilist[a] = [b]
+        else:
+            trilist[a].append(b)
+    return trilist
+def new_data(old_data):
+    word_list = ''
+    with open(filename, 'r') as file_name:
+        translator = str.maketrans('','', string.punctuation)
+        for lines in file_name.readlines():
+            words = lines.translate(translator)
+            word_list+=words
+    list_of_words = word_list.split()
+    return list_of_words
+import random
+import string
+def build_new_story(story):
+    keys_first = random.choice(list(trigram_dict(words)))
```

Comment:
Confusing usage of global `words`. Better receive it as a function input parameter.